test/iidm/ExtensionTest.cpp

```diff
@@ -0,0 +1,125 @@
+
+    const std::type_index& getType() const override;
+
+public:
+    explicit LoadExtension(Load& load, double value);
+
+    ~LoadExtension() override = default;
+
+    double getValue() const;
+
+private:  // Extension
+    void assertExtendable(const stdcxx::Reference<Extendable>& extendable) const override;
+
+private:
+    double m_value;
+};
+
+LoadExtension::LoadExtension(Load& load, double value) :
+    Extension(load),
+    m_value(value) {
+}
+
+void LoadExtension::assertExtendable(const stdcxx::Reference<Extendable>& extendable) const {
+    if (extendable && !stdcxx::isInstanceOf<Load>(extendable.get())) {
+        throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable.get()), stdcxx::demangle<Load>()));
+    }
+}
+
+const std::string& LoadExtension::getName() const {
+    static std::string s_name = "loadExtension";
+    return s_name;
+}
+
+const std::type_index& LoadExtension::getType() const {
+    static std::type_index s_type = typeid(LoadExtension);
+    return s_type;
+}
+
+double LoadExtension::getValue() const {
+    return m_value;
+}
+
+class BatteryExtension : public Extension {
+public:  // Extension
+    const std::string& getName() const override;
+
+    const std::type_index& getType() const override;
+
+public:
+    explicit BatteryExtension(Battery& battery);
+
+    ~BatteryExtension() override = default;
+
+private:  // Extension
+    void assertExtendable(const stdcxx::Reference<Extendable>& extendable) const override;
+};
+
+BatteryExtension::BatteryExtension(Battery& battery) :
+    Extension(battery) {
+}
+
+const std::string& BatteryExtension::getName() const {
+    static std::string s_name = "batteryExtension";
+    return s_name;
+}
+
+const std::type_index& BatteryExtension::getType() const {
+    static std::type_index s_type = typeid(BatteryExtension);
+    return s_type;
+}
+
+void BatteryExtension::assertExtendable(const stdcxx::Reference<Extendable>& extendable) const {
+    if (extendable && !stdcxx::isInstanceOf<Battery>(extendable.get())) {
+        throw AssertionError(stdcxx::format("Unexpected extendable type: %1% (%2% expected)", stdcxx::demangle(extendable.get()), stdcxx::demangle<Battery>()));
+    }
+}
+
+BOOST_AUTO_TEST_CASE(getExtensionByName) {
+    Network network = powsybl::network::EurostagFactory::createTutorial1Network();
+    Load& load = network.getLoad("LOAD");
+    load.addExtension(stdcxx::make_unique<LoadExtension>(load, 1.1));
+    const Load& cLoad = load;
+
+    BOOST_CHECK(!cLoad.getExtensionByName<LoadExtension>("test"));
+    BOOST_CHECK(!load.getExtensionByName<LoadExtension>("test"));
+
+    BOOST_CHECK(!cLoad.getExtensionByName<BatteryExtension>("test"));
+    BOOST_CHECK(!load.getExtensionByName<BatteryExtension>("test"));
+
+    auto cExt = cLoad.getExtensionByName<LoadExtension>("loadExtension");
+    auto ext = load.getExtensionByName<LoadExtension>("loadExtension");
```

Comment:
I would use LoadFooExt and LoadBarExt that have been created for unit testing. This would reduce the amount of code needed to test the new feature. What do you think?

Reply:
Agree! Thanks for the tip! I thought about using LoadMockExt and LoadQuxExt which are in tests but they were in submodule `iidm::converter::xml::extension`, so using them from iidm was not very conventional...